### PR TITLE
Abc widget factory docregistry dispose signal(5843)

### PR DESCRIPTION
**Pull request by ggbhat**
## References

[#5843](https://github.com/jupyterlab/jupyterlab/issues/5843)

## Code changes
Added clear signal data on dispose

## User-facing changes

No
## Backwards-incompatible changes
No

### CODE REVIEW

File: packages/rendermime/style/base.css [hunk 1]
```diff
@@ -22,6 +22,7 @@
   border: none;
   margin: 0px;
   padding: 0px;
+  line-height: normal;
```
Comment (jasongrout) [inline packages/rendermime/style/base.css]: This change is a separate issue. Can you remove it from this PR?

**Comment by jasongrout** [comment]
Thanks! The code looks good to me. Can you remove the unrelated CSS fix (that belongs on a separate PR)?

File: packages/rendermime/style/base.css (hunk 1 above)
Comment (jasongrout) [reply packages/rendermime/style/base.css]: (Looks like this CSS change is from #7832.)

Comment (ggbhat) [reply packages/rendermime/style/base.css]: Yes, not sure how it happened.

Comment (ggbhat) [reply packages/rendermime/style/base.css]: > (Looks like this CSS change is from #7832.)

I created branch out of my previous branch:) now I removed and checked in. Kindly review.

**Pull request merged**